httpx/client.py

```diff
@@ -47,7 +48,11 @@
     ResponseContent,
     URLTypes,
 )
-from .utils import get_netrc_login
+from .utils import get_environment_proxies, get_netrc_login
+
+ProxiesTypes = typing.Union[
+    URLTypes, AsyncDispatcher, typing.Dict[str, typing.Union[URLTypes, AsyncDispatcher]]
+]
```

Comment:
This should probably go in `models.py` along with the other reusable types.

Reply:
Can't go there because of the `AsyncDispatcher` type :/

Reply:
Would using the if typing.TYPE_CHECKING trick and forward-referencing AsyncDispatcher (ie putting it in quotes in the type) solve it? https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles

See also this discussion: https://github.com/encode/httpx/pull/305#discussion_r321125214